### PR TITLE
fix: improve UI

### PR DESCRIPTION
**Pull request by Hitesh-Kandpal**
### Pre-requisites

- [x] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](../../CONTRIBUTING.md#submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
- [x] I have performed preliminary testing to ensure that any existing features are not impacted and any new features are working as expected.

### PR Details

PR details have been updated as per the given format (see below)

- [x] PR title adheres to the format specified in guidelines (e.g., `feat: add admin login page`)
- [x] Description has been added
- [x] Related changes have been added (optional)
- [x] Screenshots have been added (optional)
- [ ] Pending actions have been added (optional)
- [x] Any other additional notes have been added (optional)

### Additional Information

- [x] Appropriate label(s) have been added (`ready for review` should be added if the PR is ready to be reviewed)
- [x] Assignee(s) and reviewer(s) have been added (optional)

---

**Description:**

Did some changed in UI and used better components and styling, also added a title to page, and instead of local storage used session storage

**Related changes:**

1. in transcript-analyzer component used primeng file upload component and did the necessary changes in all files
2. did one change in file-processor component scss file
3. added main title in index.html and added label in en.json file
4. used session storage instead of local storage

**Screenshots:**

Screenshot 2024-05-14 at 7 27 49 PM

https://github.com/OsmosysSoftware/osm-transcript-summarizer/assets/168435416/f28c8f49-e53a-48c5-956e-2f19e1507808

**Pending actions:**

N/A

**Additional notes:**

The reason for using session storage over local storage was that while I was continuously uploading files and then deleting them from the database for testing purposes, local storage was accumulating job IDs that had been deleted from the database. Additionally, since there is a limit of 10 in the status API for fetching statuses, the local storage, which includes old job IDs, makes it difficult to retrieve the status of recently uploaded files. Therefore, using session storage makes more sense in this scenario.

## Summary by CodeRabbit

- **New Features**
  - Added a file upload component in the transcript analyzer with updated event handling and styles.
  - Updated i18n translations with new keys for file upload instructions.

- **Improvements**
  - Changed storage for job IDs from `localStorage` to `sessionStorage` for improved session management.

- **Bug Fixes**
  - Fixed file download functionality to correctly replace specific file extensions with `.md`.

- **Configuration**
  - Updated Redis Docker port in `.env.example` to `6379`.

- **Refactor**
  - Simplified file handling logic in the transcript analyzer component.

### CODE REVIEW

File: apps/portal/src/app/features/file-processor/file-processor.component.scss [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .table-container {
```
Comment (xixas) [inline apps/portal/src/app/features/file-processor/file-processor.component.scss:1]: Add title heading to the page not just browser title.
Add what files are accepted and max size of the file.

Comment (xixas) [reply apps/portal/src/app/features/file-processor/file-processor.component.scss:1]: Use different color of SUmmarize me(like Success)

File: apps/portal/src/app/features/file.service.ts [hunk 2]
```diff
@@ -16,12 +16,12 @@ export class FileService {
     private graphqlService: GraphqlService,
     private apollo: Apollo,
   ) {
-    this.jobIdsSubject.next(JSON.parse(localStorage.getItem('jobIds') || '[]'));
+    this.jobIdsSubject.next(JSON.parse(sessionStorage.getItem('jobIds') || '[]'));
```
Comment (xixas) [inline apps/portal/src/app/features/file.service.ts:19]: When are job ids removed in this case?

**Review by xixas — CHANGES_REQUESTED** [review]

File: apps/portal/src/app/features/file-processor/file-processor.component.scss (hunk 1 above)
Comment (Hitesh-Kandpal) [reply apps/portal/src/app/features/file-processor/file-processor.component.scss:1]: 1. added title heading for page
2. used primeNg p-message to show info valid file types and max size allowed
3.used success for summarize file

(also did some changes after adding p-message to make UI consistent)

File: apps/portal/src/app/features/file.service.ts (hunk 2 above)
Comment (Hitesh-Kandpal) [reply apps/portal/src/app/features/file.service.ts:19]: as soon as user closed the tab Job Ids are removed from storage

**Review by xixas — APPROVED** [review]

File: apps/portal/src/app/features/file-processor/file-processor.component.ts [hunk 3]
```diff
@@ -134,7 +134,7 @@ export class FileProcessorComponent implements OnInit, OnDestroy {
   downloadFile(summary: string | null | undefined, fileName: string): void {
     if (summary !== undefined && summary !== null) {
       const blob = new Blob([summary], { type: 'text/markdown' });
-      FileSaver.saveAs(blob, `${fileName}.md`);
+      FileSaver.saveAs(blob, `${fileName.replace(/\.txt|\.vtt/, '')}.md`);
```
[the remark this answers is not in the document]
Comment (xixas) [reply apps/portal/src/app/features/file-processor/file-processor.component.ts:137]: Remove otherExtensions just keep txt and vtt(upper and lower case both) Basically remove file extension

**Review by xixas — CHANGES_REQUESTED** [review]

File: apps/portal/src/app/features/file-processor/file-processor.component.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (xixas) [reply apps/portal/src/app/features/file-processor/file-processor.component.ts:137]: Can you commit it?

**Review by xixas — APPROVED** [review]

**Review by xixas — APPROVED** [review]

**Pull request merged**